astropy/io/fits/tests/test_image.py

```diff
@@ -1725,3 +1725,12 @@ def test_bzero_mishandled_info(tmpdir):
     hdu.writeto(filename, clobber=True)
     hdul = fits.open(filename)
     hdul.info()
+
+
+def test_image_write_readonly(tmpdir):
+    # Regression test to make sure that we can write out read-only arrays (#5512)
+    x = np.array([1,2,3])
+    x.setflags(write=False)
+    ghdu = fits.ImageHDU(data=x)
+    ghdu.add_datasum()
+    ghdu.writeto(tmpdir.join('test.fits').strpath)
```

Comment:
If I try and write a CompImageHDU I get an error ``RuntimeError: writing to compressed image is not supported`` - would you have time to write a unit test and do a PR to my branch with what you had in mind?